### PR TITLE
api v1 - OAuth user mapping

### PR DESCRIPTION
**Pull request by pitr-ch**

### CODE REVIEW

File: lib/foreman/default_settings/loader.rb [hunk 1]
```diff
@@ -65,7 +65,8 @@ def load(reset=false)
 
             [ set('oauth_active', "Should foreman use OAuth for authorization in API", false),
               set('oauth_consumer_key', "OAuth consumer key", 'katello'),
-              set('oauth_consumer_secret', "OAuth consumer secret", 'shhhh')
+              set('oauth_consumer_secret', "OAuth consumer secret", 'shhhh'),
+              set('oauth_map_users', "Should foreman map users by username in request-header", true)
```
Comment (ohadlevy) [inline lib/foreman/default_settings/loader.rb:69]: can we turn off oauth by default? it seems like the defaults are way to easy to guess.

Comment (pitr-ch) [inline lib/foreman/default_settings/loader.rb:69]: yes I agree, it's turned off, see line 66 :)

**Comment by ohadlevy** [comment]
my bad, what about tests? :)

**Comment by pitr-ch** [comment]
on my todo

**Pull request closed without merging**